x/ccv/consumer/ibc_module.go

```diff
@@ -270,18 +270,25 @@ func (am AppModule) OnAcknowledgementPacket(
 }
 
 // OnTimeoutPacket implements the IBCModule interface
+// the CCV channel state is changed to CLOSED
+// by the IBC module as the channel is ORDERED
 func (am AppModule) OnTimeoutPacket(
 	ctx sdk.Context,
 	packet channeltypes.Packet,
 	_ sdk.AccAddress,
 ) error {
-	var data ccv.SlashPacketData
-	if err := ccv.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
-		return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal consumer packet data: %s", err.Error())
-	}
+	var slashData ccv.SlashPacketData
+	var vscMaturedData ccv.VSCMaturedPacketData
 
-	if err := am.keeper.OnTimeoutPacket(ctx, packet, data); err != nil {
-		return err
+	err := ccv.ModuleCdc.UnmarshalJSON(packet.GetData(), &slashData)
```

Comment:
This could still accidentally unmarshal one type as another in some cases if in the future we changed the field names or something. Probably not a problem right now, but pretty sketchy. We probably need to think of a concrete way to differentiate these packets types, other than the lucky fact that they might not be able to deserialize into each other for now. Is there already something somewhere else in the codebase? @mpoke

Reply:
This function seemed like a sanity check and nothing more. It unmarshals to see if the type is correct and outputs an event if it is, otherwise it errors.

Reply:
I agree that we don't need to unmarshal here.

 Regarding the more general question about differentiating between different packet types, we do the same on the provider side, i.e., https://github.com/cosmos/interchain-security/blob/f6cec81f2b7022021698cf09c45d928dfd8f85a0/x/ccv/provider/ibc_module.go#L170. The easiest way would be to add a packet type in `ccv.proto`.